src/page.js

```diff
@@ -6,6 +6,28 @@ export default class Page {
         this.target = 'page$' + pageId;
         this.phantom = phantom;
     }
+
+    on(event, runOnPhantom, callback) {
```

Comment:
I see what you mean. Because you do `listeners.otherListeners.push` it adds to the chain of events. Right?